### ISSUE TITLE
MMD疑问

### ISSUE THREAD

**Issue by weixifei6688** [jindongwang/transferlearning#143]

**Comment by weixifei6688** [comment]
线性核MMD：
loss = torch.mean((delta[:-1] * delta[1:]).sum(1))（第一种：内积计算不理解为什么这么计算？）
loss = torch.mean(torch.mm(delta, torch.transpose(delta, 0, 1)))（第二种）
loss=np.mean(delta.dot(delta.T))（我的理解，通过内积计算）
王老师您好，十分感谢您的资料，在我学习的过程中帮助巨大！
我有些疑问想请教您，第一种MMD计算我不太理解为什么那么计算，我经过测试这两种方法结果是不一样的（我的计算方式和第二种方法是一样的结果，第一第二种都是您给的参考代码），这样的计算方式不仅出现在线性核当中，多项式核也出现了这样的计算方式，我用纯粹的Python语言通过内积计算了多项式核，结果也是不一样的，希望你能给我答疑解惑（刚开始入门学习迁移学习，可能有些不太懂，望您理解。）
多项式核我的计算方式（一些参数我随意设置的）：
XX=np.mean(pow(diff_1.dot(diff_1.T)+2,2))
YY=np.mean(pow(diff_2.dot(diff_2.T)+2,2))
XY=np.mean(pow(diff_1.dot(diff_2.T)+2,2))
YX=np.mean(pow(diff_2.dot(diff_1.T)+2,2))
loss=XX+YY-XY-YX
（我的理解，和上面的线性核是一样的道理）

**Comment by jindongwang** [comment]
@weixifei6688 感谢指正！你的理解是正确的，确实是我们的代码有问题，我已修改linear_mmd的代码，请见最新的提交。因为一般都在用rbf的kernel，所以linear用的少，没及时发现。也请你参与此项目，修改mmd.py中的poly_mmd相关代码！

**Comment by weixifei6688** [comment]
王老师，您好，我按照自己理解在您代码的基础上修改了多项式核的代码，我现在把他通过邮件发给您（GitHub我刚开始使用，不太熟悉），若有问题，望您更正（我使用的框架是TensorFlow，对pytorch不太熟悉，迁移学习也是刚刚入门学习），再次感谢您对于迁移学习资源的共享！
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;










&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;

**Comment by weixifei6688** [comment]
> @weixifei6688 感谢指正！你的理解是正确的，确实是我们的代码有问题，我已修改linear_mmd的代码，请见最新的提交。因为一般都在用rbf的kernel，所以linear用的少，没及时发现。也请你参与此项目，修改mmd.py中的poly_mmd相关代码！

**Comment by jindongwang** [comment]
@weixifei6688 Good work! 感谢！我稍后将其合并到代码库中！
